runtime/jcl/common/sun_reflect_ConstantPool.c

```diff
@@ -267,11 +270,21 @@ getMethodAt(JNIEnv *env, jobject constantPoolOop, jint cpIndex, UDATA resolveFla
 				methodID = (jmethodID) vmFunctions->getJNIMethodID(vmThread, method);
 			}
 		}
-		vmFunctions->internalReleaseVMAccess(vmThread);
-	}
 
-	if (NULL != methodID) {
-		returnValue = (*env)->ToReflectedMethod(env, constantPoolOop, methodID, J9CPTYPE_STATIC_METHOD == cpType);
+		if (NULL != methodID) {
+			j9object_t classObject = J9CLASS_FROMJ9CLASSREF(vmThread, constantPoolOop)->classObject;
+			jclass jlClass = vmFunctions->j9jni_createLocalRef(env, classObject);
+			if (NULL != jlClass) {
+				vmFunctions->internalReleaseVMAccess(vmThread);
+				returnValue = (*env)->ToReflectedMethod(env, jlClass, methodID, J9CPTYPE_STATIC_METHOD == cpType);
+				vmFunctions->internalEnterVMFromJNI(vmThread);
```

Comment:
There's an lot of atomic operations happening here - 
* this method drops VMAccess, 
* then toReflectMethod will acquire and drop it again, 
* then this method acquires it so it can be dropped again at the bottom of the method.

 Things that we can ignore: 
* deleting the localref isn't necessary - it will be cleaned up when the method returns.
* extra acquire / releases in the setNativeOOM case are fine.  In fact, use `throwNativeOOMError()` as it handles the acquire / release for you.